game/functions/core/client/statusEffect/fn_statusEffect_postInit.sqf

```diff
@@ -17,3 +17,26 @@ private _fnc_setCoreStatusEffects = {
 
 call _fnc_setCoreStatusEffects;
 player addEventHandler ["Respawn", _fnc_setCoreStatusEffects];
+
+vgm_c_statusEffect_eachFrameEH = addMissionEventHandler ["EachFrame", {
+    // Queue of end times for status effect reasons, sorted by end time.
+    private _endTimesQueue = player getVariable ["vgm_c_statusEffect_endTimesQueue", []];
+    if (_endTimesQueue isEqualTo []) exitWith {};
+
+    // First item in the queue will always be the soonest.
+    _endTimesQueue # 0 params ["_nextEndTime", "_effectDetails"];
+
+    if (serverTime < _nextEndTime) exitWith {};
```

Comment:
I would use `time` which is tracked since mission started and affected by accTime in SP (does not matter for us though).
You usually want to avoid server time as it is more prone to floating point errors.

```suggestion
    if (time < _nextEndTime) exitWith {};
```